client/my-sites/domains/domain-management/dns/mx-record.jsx

```diff
@@ -14,56 +15,71 @@ import FormTextInput from 'components/forms/form-text-input';
 import FormTextInputWithAffixes from 'components/forms/form-text-input-with-affixes';
 import FormSettingExplanation from 'components/forms/form-setting-explanation' ;
 
-const MxRecord = React.createClass( {
-	statics: {
-		initialFields: {
-			name: '',
-			data: '',
-			aux: 10
-		}
-	},
+class MxRecord extends React.PureComponent {
+	static initialFields = {
+		name: '',
+		data: '',
+		aux: 10
+	};
 
-	propTypes: {
+	static propTypes = {
 		fieldValues: React.PropTypes.object.isRequired,
 		onChange: React.PropTypes.func.isRequired,
 		selectedDomainName: React.PropTypes.string.isRequired,
 		show: React.PropTypes.bool.isRequired
-	},
-
-	fieldNames: [ 'name', 'data', 'aux' ],
+	};
 
 	render() {
```

Comment:
Not sure what's special about this component, but when I select `MX` from the list, I get this warning in the console:
```
Warning: FormTextInput is changing an uncontrolled input of type text to be controlled. Input elements should not switch from uncontrolled to controlled (or vice versa). Decide between using a controlled or uncontrolled input element for the lifetime of the component. More info: https://fb.me/react-controlled-components
```
(it might have been like that before your refactor, though)

Edit: seems `SRV` is the same 🤔 But not A, CNAME, TXT.